File: pokemon-website/src/Types/PokemonResponse.ts
type PokemonResponse = {
	name: string;
	imageUrl: string;
	types: PokemonType[];
	description: string;
	health: number;
	attack: number;
	defense: number;
	specialAttack: number;
	specialDefense: number;
	speed: number;
};

type PokemonType = {
	name: string;
	color: string;
};

File: pokedex-backend/src/main/java/obss/pokemon/model/pokemon/PokemonResponse.java
package obss.pokemon.model.pokemon;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import obss.pokemon.model.pokemonType.PokemonTypeResponse;

import java.util.Set;

@NoArgsConstructor
@Getter
@Setter
public class PokemonResponse {
    private String name;
    private Set<PokemonTypeResponse> types;
    private String imageUrl;
    private String description;
    private int health;
    private int attack;
    private int defense;
    private int specialAttack;
    private int specialDefense;
    private int speed;
}
